iib/workers/tasks/utils.py

```diff
@@ -1191,19 +1185,27 @@ def prepare_request_for_build(
 
     binary_image = build_request_config.binary_image(request_index, distribution_scope)
     if binary_image == "scratch":  # binaryless image mode
+        arches = set(["linux-amd64"])
         binary_image_resolved = binary_image
         binary_image_arches = arches
     else:
+        arches = gather_index_image_arches(build_request_config, index_info)
         binary_image_resolved = get_resolved_image(binary_image)
         binary_image_arches = get_image_arches(binary_image_resolved)
 
+    if not arches:
+        raise IIBError('No arches were provided to build the index image')
+
     if not arches.issubset(binary_image_arches):
         raise IIBError(
             'The binary image is not available for the following arches: {}'.format(
                 ', '.join(sorted(arches - binary_image_arches))
             )
         )
 
+    arches_str = ', '.join(sorted(arches))
```

Comment:
Why are we not returning this format (joined and sorted array) from the method `gather_index_image_arches` method? Also, this exception "No arches were provided to build the index image" is already thrown within the mentioned method, should we deduplicate that?

Reply:
We use this joined format only for printing it out. We should not return it because we work with arches as set. This way you don't have to parse it from string when we work with arches later on.